OvaryModelGenerator/OvaryModelGenerator.h

```diff
@@ -8,20 +8,24 @@
 class OvaryModelGenerator
 {
     public:
-        int nslices, nrot;
-        double d, h, w;
-        std::string outdirstr;
-
-        OvaryModelGenerator(int slice, int rot, double de, double he, double wi, std::string outdir);
+        OvaryModelGenerator(int slice, int rot, double de, double he, double wi);
         ~OvaryModelGenerator();
-        OvaryModelGenerator(const OvaryModelGenerator &other);
+        OvaryModelGenerator(const OvaryModelGenerator &other) = delete;
+        OvaryModelGenerator & operator= (const OvaryModelGenerator & other) = delete;
         // static int inputCheck();
         void SetRotationalSlices(int rot);
         void SetOutDir(std::string out);
         void SetLongAxisSlices(int slice);
         void SetDimensions(double de, double he, double wi);
         void Generate();
-        void GetOutput(std::vector<vtkSmartPointer<vtkPolyData>> vec, int imax);
+        std::vector<vtkSmartPointer<vtkPolyData>> &GetOutput();
```

Comment:
This is a design change. To follow the "Single Responsibility Principle" in SOLID principles, we don't want a class to more things than they should. Here, generating a model is one thing, exporting them to files is another, and they should be designed into separate modules.  This allows a better cohesion within the module and decoupled it from the exporting logic. 

In this change, instead of passing a vector to getouput for exporting, we simply just save it as internal data. And GetOutput() now just returns a reference to this data.